pkg/yurttunnel/handlerwrapper/tracerequest/tracereq.go

```diff
@@ -75,32 +88,37 @@ func (trm *traceReqMiddleware) WrapHandler(handler http.Handler) http.Handler {
 
 		req.URL.Scheme = scheme
 		req.URL.Host = req.Host
-
-		host, port, err := net.SplitHostPort(req.Host)
-		if err != nil {
-			klog.Errorf("request host(%s) is invalid, %v", req.Host, err)
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
-		}
-
-		// host for accessing edge component(like kubelet) is hostname, not node ip.
-		if ip := net.ParseIP(host); ip == nil {
-			// 1. transform hostname to nodeIP for request in order to send request to nodeIP address at tunnel-agent
-			// 2. put hostname into X-Tunnel-Proxy-Host request header in order to select the correct backend agent.
-			if err := trm.modifyRequest(req, host, port); err != nil {
+		if completed, err := trm.handleRequestsFromKAS(req); completed || err != nil {
```

Comment:
Should it be `!completed || err != nil` ?
And why we need to add func `handleRequestsFromKAS`?

Reply:
@Congrool a comment has been added to introduce the reason of adding `handleRequestsFromKAS`.
and the if condition should be `completed || err != nil`  because we can go through to `ServeHTTP` after `handleRequestsFromKAS ` is completed.